pytrip/tripexecuter2/plan.py

```diff
@@ -0,0 +1,293 @@
+#    the Free Software Foundation, either version 3 of the License, or
+#    (at your option) any later version.
+#
+#    PyTRiP98 is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU General Public License for more details.
+#
+#    You should have received a copy of the GNU General Public License
+#    along with PyTRiP98.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+Structure:
+A Plan() may hold one or multiple Field() objects. When a Plan() is proper setup, it can be executed
+with methods in Execute().
+"""
+
+import os
+import uuid
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class Plan():
+    def __init__(self, basename="", comment=""):
+        """
+        A plan Object, which may hold several fields, general setup, and possible also output,
+        if it has been calculated.
+        :params str basename: TRiP98 qualified plan name. E.g. "test000001"
+        :params str comment:" any string for documentation purposes.
+        """
+
+        self.__uuid__ = uuid.uuid4()  # for uniquely identifying this plan
+        self.basename = basename
+        self.comment = comment
+
+        self.fields = []  # list of Field() objects
+        self.voi_target = None  # put target Voi() here. (Not needed if self.target_dose_cube is set (incube))
+        self.vois_oar = []  # list of Voi() objects which are considered as OARs
+
+        self.dosecubes = []  # list of DosCube() objects (i.e. results)
+        self.letcubes = []  # list of LETCubes()
+
+        # remote planning
+        self.remote = False  # remote or local execution
+        self.servername = ""
+        self.username = ""
+        self.password = ""
+
+        # directories and file paths.
+        self.working_dir = ""  # directory where all input files are stored, and where all output files will be put.
+        self.ddd_dir = "$TRIP98/DATA/DDD/12C/RF3MM/12C.*"
+        self.spc_dir = ""
+        self.sis_path = ""
+        self.hlut_path = "$TRIP98/DATA/19990211.hlut"
+        self.dedx_path = "$TRIP98/DATA/DEDX/20040607.dedx"
+
+        self.res_tissue_type = ""
+        self.target_tissue_type = ""
+        self.active = False
+        self.optimize = True  # switch wheter it should be optimized. If False, then a RST file will be loaded instead.
+        self.iterations = 500
+        self.opt_method = "phys"
+        self.opt_principle = "H2Obased"
+        self.opt_alg = "cg"
+        self.dose_alg = "cl"
+        self.bio_alg = "cl"
+        self.eps = 1e-3
+        self.geps = 1e-4
+
+        # algorithms for dose optimizations
+        self._opt_principles = {"H2Obased": "Very simplified single field optimization",
+                                "CTbased": "Full optimization, multiple fields"}
+
+        self._opt_methods = {"phys": "Physical dose only",
```

Comment:
All this dictionaries could be removed from `__init__` and moved to a class fields as the are constants.